### PR TITLE
Use a notification driver with the new channel extender

### PR DESCRIPTION
**Pull request by SychO9**
**Part Of flarum/core#2432**

Hmm, do we want to create a job for it and queue it ? it used to be executed inside the `SendNotificationJob` job.

### CODE REVIEW

**Comment by askvortsov1** [comment]
> Hmm, do we want to create a job for it and queue it ? it used to be executed inside the `SendNotificationJob` job.

I'd lean towards yes on this one, but @flarum/core might disagree. Lets wait for a decision on flarum/core#2432 first though.

**Comment by tankerkiller125** [comment]
I'm going to say Queue it, Pretty much my personal opinion is to always queue things unless it's critical to execute immediately. Notifications of any kind in my opinion are always queueable.

**Comment by clarkwinkelmann** [comment]
Queue definitely make sense. Technically we could implement `NotificationDriverInterface::send` to always be called from a queue in core for all notifications... But would we want that? Though I think this might create an issue if an extension actually needs to call its code without a queue.

Other question is one job per notification, or one job that loops through the users?

**Comment by SychO9** [comment]
> Queue definitely make sense. Technically we could implement `NotificationDriverInterface::send` to always be called from a queue in core for all notifications... But would we want that? Though I think this might create an issue if an extension actually needs to call its code without a queue.

Yea I don't think we'd want to force it on a queue.

> Other question is one job per notification, or one job that loops through the users?

I'd say one job that loops through the users.

File: src/PusherNotificationDriver.php [hunk 1]
```diff
@@ -11,36 +11,32 @@
...
      * {@inheritDoc}
      */
     public function send(BlueprintInterface $blueprint, array $users): void
     {
-        foreach ($users as $user) {
-            if ($user->shouldAlert($blueprint::getType())) {
-                $this->pusher->trigger('private-user'.$user->id, 'notification', null);
-            }
-        }
+        $this->queue->push(new SendPusherNotificationsJob($blueprint, $users));
```
Comment (askvortsov1) [inline src/PusherNotificationDriver.php]: Perhaps we should only do this if users has length like with Email and Alerts?

**Review by askvortsov1 — CHANGES_REQUESTED** [review]

File: src/PusherNotificationDriver.php (hunk 1 above)
Comment (SychO9) [reply src/PusherNotificationDriver.php]: ah yes you're right, no need to queue a job that won't do anything.

**Review by askvortsov1 — APPROVED** [review]

**Review by tankerkiller125 — APPROVED** [review]

**Pull request merged**